Simplify DataFlash initialisation

peterbarker opened the pull request.

OXINARF commented on libraries/DataFlash/DataFlash.h:
```diff
@@ -75,7 +75,6 @@ class DataFlash_Class
 
     // possibly expensive calls to start log system:
-    bool NeedPrep();
     void Prep();
```
Nitpicking: does the Prep still needs to be public? I would say it could even be removed and the one liner incorporated into *init*.

OXINARF commented on libraries/DataFlash/DataFlash.h:
```diff
@@ -286,4 +285,6 @@ class DataFlash_Class
 
     void Log_Write_EKF_Timing(const char *name, uint64_t time_us, const struct ekf_timing &timing);
+
+    bool NeedPrep();
```
If this is private what is it needed for? I didn't see it being used anywhere in DataFlash_Class.

peterbarker commented:
On Tue, 13 Jun 2017, Francisco Ferreira wrote:

> I really like that you are simplifying the DataFlash APIs. I've made two

Long overdue :-)

I'm planning on dripping in several more changes.  Most notably I want to 
move should_log into the dataflash library.

> Nitpicking: does the Prep still needs to be public? I would say it could
> even be removed and the one liner incorporated into init.

It does not.  I moved Prep() in rather late.  If things settle we might 
get rid of it entirely.

> +    bool NeedPrep();
> 
> If this is private what is it needed for? I didn't see it being used
> anywhere in DataFlash_Class.

Removed.

OXINARF commented:
Merged, thanks!

By the way, one thing I forgot to mention before is that it is interesting that Rover and Tracker do *start_logging* on *log_init* but the other vehicles don't. It's possibly another place to cleanup.
